src/imap/mod.rs

```diff
@@ -314,9 +314,15 @@ impl Imap {
     }
 
     async fn unsetup_handle(&mut self, context: &Context) {
+        // Close folder if messages should be expunged
+        if let Err(err) = self.close_folder(context).await {
+            warn!(context, "failed to close folder: {:?}", err);
+        }
```

Comment:
what is the timeout here? i.e. if unsetup handle is called but the network is gone, how long does it take?